pkg/core/mpt/trie.go

```diff
@@ -422,8 +422,20 @@ func (t *Trie) Flush() {
 	}
 }
 
+func IsActiveValue(v []byte) bool {
+	return len(v) > 4 && v[len(v)-5] == 1
+}
+
+func getFromStore(key []byte, mode TrieMode, store *storage.MemCachedStore) ([]byte, error) {
+	data, err := store.Get(key)
+	if err == nil && mode.GC() && !IsActiveValue(data) {
```

Comment:
This prevents all users of `getFromStore` from getting available historical state of a node that has been removed in the latest version of MPT, e.g. `findstates` and `getproof` RPC calls.

On the contrary, we should adjust Billet's `(b *Billet) GetFromStore(h util.Uint256)` with regarding to the `active` byte, so that inactive items that are not traceable but still exist in the storage won't be returned from this method.

Reply:
I think it's a program error if we're to give these nodes to the outside users. We mostly use `Billet` for traversing to collect a reply for `getMPTData` request and imagine the situation: some stale node is asking for stale MPT nodes that are marked as `inactive`, lower than P2 and still in storage. We shouldn't reply to such requests.

Reply:
What's bad about returning them? While we can separate "inactive, but traceable" from "useless garbage" (that's what GC does in its essence) it requires knowing current height and that's a moving target anyway (one GetMPTData at height `h`, another at height `h+1`), so if a peer requests some stale data that's his responsibility, if we have it --- it's not hard to return it, if we don't --- sorry. It's kinda the same for untraceable blocks, fresh nodes will request them without knowing whether the peer has them (tail-stripping nodes won't).